kbfsgit/runner.go

```diff
@@ -376,30 +376,47 @@ func (r *runner) printJournalStatus(
 		r.errput.Write([]byte(str))
 	}
 
-	ticker := time.NewTicker(1 * time.Second)
+	ticker := time.NewTicker(1 * time.Millisecond)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
-		case <-doneCh:
-		}
-		status, err := jServer.JournalStatus(tlf)
-		if err != nil {
-			r.log.CDebugf(ctx, "Error getting status: %+v", err)
-			return
-		}
+			status, err := jServer.JournalStatus(tlf)
+			if err != nil {
+				r.log.CDebugf(ctx, "Error getting status: %+v", err)
+				return
+			}
 
-		if r.verbosity >= 1 && r.progress {
-			eraseStr := strings.Repeat("\b", lastByteCount)
-			flushed := firstStatus.UnflushedBytes - status.UnflushedBytes
-			str := fmt.Sprintf(
-				bytesFmt, percent(flushed, firstStatus.UnflushedBytes),
-				humanizeBytes(flushed, firstStatus.UnflushedBytes))
-			lastByteCount = len(str)
-			r.errput.Write([]byte(eraseStr + str))
-		}
+			if r.verbosity >= 1 && r.progress {
+				eraseStr := strings.Repeat("\b", lastByteCount)
+				flushed := firstStatus.UnflushedBytes - status.UnflushedBytes
+				str := fmt.Sprintf(
+					bytesFmt, percent(flushed, firstStatus.UnflushedBytes),
+					humanizeBytes(flushed, firstStatus.UnflushedBytes))
+				lastByteCount = len(str)
+				r.errput.Write([]byte(eraseStr + str))
+			}
+
+			if status.UnflushedBytes == 0 {
```

Comment:
Do we still need this block of code?  Seems like we can assume `doneCh` will be closed.

Reply:
Hmmm this is also used for cloning, which passes in a nil `doneCh`. Or are you suggesting to change that part to use channel as well?

Reply:
I don't see anywhere that passes in nil to printJournalStatus. Certainly not handleClone, which is copying data out of KBFS and into the local filesystem. What are you referring to?

Reply:
Oh sorry! I was confusing this with the added `fsEvents` in [printJournalStatusUntilFlushed ](https://github.com/keybase/kbfs/pull/1216/files#diff-b6999c82a099222a06e9532d0ff7e054R624). I'm implement your suggestion then. Thanks